bindings/aws/s3/s3.go

```diff
@@ -152,40 +153,39 @@ func (s *AWSS3) create(ctx context.Context, req *bindings.InvokeRequest) (*bindi
 	if err != nil {
 		return nil, fmt.Errorf("s3 binding error: error merging metadata: %w", err)
 	}
-	var key string
-	if val, ok := req.Metadata[metadataKey]; ok && val != "" {
-		key = val
-	} else {
-		key = uuid.New().String()
-		s.logger.Debugf("s3 binding: key not found. generating key %s", key)
-	}
 
-	d, err := strconv.Unquote(string(req.Data))
-	if err == nil {
-		req.Data = []byte(d)
+	key := req.Metadata[metadataKey]
+	if key == "" {
```

Comment:
I wonder if the following _if_ block should be extracted to its own function so it can be properly unit tested. On test scenario that came to me is the one where the `key` is just spaces. Would that be valid and/or appropriate in the context of S3?

Reply:
IMHO this is one of those cases in which we should just accept whatever the user passes and if that's invalid, the service will complain. Otherwise, next we'd find ourselves dealing with users trying to send UTF-8 sequences and having to address that case too, where maybe one S3 implementation supports them and the other one doesn't.

(Also note that the behavior hasn't changed, I just simplified the check since retrieving a value that doesn't exist from a map returns the zero value automatically, in this case the empty string)

Reply:
Yeah I agree.